lib/vrl/stdlib/src/del.rs

```diff
@@ -166,14 +167,23 @@ impl Expression for DelFn {
     }
 
     fn type_def(&self, _: &state::Compiler) -> TypeDef {
-        TypeDef::new().unknown()
+        TypeDef::any()
     }
 
     fn update_state(
         &mut self,
         state: &mut state::Compiler,
     ) -> std::result::Result<(), ExpressionError> {
-        self.query.delete_type_def(state);
+        // FIXME(Jean): This should also delete non-external queries, as `del(foo.bar)` is
+        // supported.
+        if self.query.is_external() {
+            match self.query.delete_type_def(state) {
+                Err(remove::Error::RootPath)
+                | Err(remove::Error::CoalescedPath)
+                | Err(remove::Error::NegativeIndexPath) => { /* TODO: handle removal errors */ }
```

Comment:
This is similar to existing behaviour, except that the `value` crate now has the "correct" behaviour, which our stdlib (for now) then ignores.

Reply:
Is this TODO left to be done in this PR, or do we have an issue for that?

Reply:
No, the TODO isn't for this PR, as there's no change in behavior. I'll create a follow-up issue.

Reply:
#11264